File: secondSeminar/src/main/java/org/sopt/secondSeminar/domain/Member.java
package org.sopt.secondSeminar.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.sopt.secondSeminar.dto.request.MemberCreateRequest;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(updatable = false)
    private Long id;
    private String name;
    private String nickname;
    private int age;

    @Embedded
    private SOPT sopt;

    @Builder
    private Member(final String name,
                   final String nickname,
                   final int age,
                   final SOPT sopt) {
        this.name = name;
        this.nickname = nickname;
        this.age = age;
        this.sopt = sopt;
    }

    public static Member create(final MemberCreateRequest request) {
        return Member.builder()
                .name(request.name())
                .nickname(request.nickname())
                .age(request.age())
                .sopt(request.sopt())
                .build();
    }

    public void updateSOPT(final SOPT sopt) {
        this.sopt = sopt;
    }
}

File: secondKotlin/src/main/kotlin/org/sopt/secondKotlin/domain/Member.kt
package org.sopt.secondKotlin.domain

import jakarta.persistence.Embedded
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import org.sopt.secondKotlin.dto.request.MemberCreateRequest

@Entity
class Member(
    var name: String,
    var nickname: String,
    var age: Int,

    @Embedded
    var SOPT: SOPT,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = 0 //추후 테스트코드 작성을 위해 우선 기본생성자로 부여할 수 있게 하겠습니다
) {
    companion object{
        fun create(request: MemberCreateRequest): Member {
            return Member(request.name, request.nickname, request.age, request.sopt)
        }
    }
}